traversal/planner/PlannerVertex.java

```diff
@@ -165,70 +157,43 @@ void recordCost() {
         costLastRecorded = costNext;
     }
 
-    void recordResults() {
-        varIsStartingVertex_result = (int) Math.round(varIsStartingVertex.solutionValue());
-        varIsEndingVertex_result = (int) Math.round(varIsEndingVertex.solutionValue());
-        varHasIncomingEdges_result = (int) Math.round(varHasIncomingEdges.solutionValue());
-        varHasOutgoingEdges_result = (int) Math.round(varHasOutgoingEdges.solutionValue());
-        assert !(isStartingVertex() && isEndingVertex());
-        assert (isStartingVertex() ^ hasIncomingEdges());
-        assert (isEndingVertex() ^ hasOutgoingEdges());
+    boolean validResults() {
+        return !(isStartingVertex() && isEndingVertex()) && (isStartingVertex() ^ hasIncomingEdges())
+                && (isEndingVertex() ^ hasOutgoingEdges());
     }
 
     void resetInitialValue() {
-        varIsStartingVertex_init = 0;
-        varIsEndingVertex_init = 0;
-        varHasIncomingEdges_init = 0;
-        varHasOutgoingEdges_init = 0;
+        varIsStartingVertex.clearInitial();
+        varIsEndingVertex.clearInitial();
+        varHasIncomingEdges.clearInitial();
+        varHasOutgoingEdges.clearInitial();
     }
 
     void setStartingVertexInitial() {
-        varIsStartingVertex_init = 1;
-        varIsEndingVertex_init = 0;
+        varIsStartingVertex.setInitial(true);
+        varIsEndingVertex.setInitial(false);
+        varHasIncomingEdges.setInitial(false);
     }
 
     void setEndingVertexInitial() {
-        varIsEndingVertex_init = 1;
-        assert varIsStartingVertex_init == 0;
-        assert varHasOutgoingEdges_init == 0;
-        assert varHasIncomingEdges_init == 1;
+        varIsEndingVertex.setInitial(true);
+        varIsStartingVertex.setInitial(false);
+        varHasOutgoingEdges.setInitial(false);
+//        assert varIsStartingVertex.getHint() == 0;
+//        assert varHasOutgoingEdges.getHint() == 0;
+        assert varHasIncomingEdges.getInitial() == 1;
     }
 
     void setHasOutgoingEdgesInitial() {
-        varHasOutgoingEdges_init = 1;
-        assert varIsEndingVertex_init == 0;
+        varHasOutgoingEdges.setInitial(true);
+        varIsEndingVertex.setInitial(false);
+//        assert varIsEndingVertex.getHint() == 0;
```

Comment:
we have to be more explicit about setting initial values now, rather than using initial (implicit) 0 values